app/actions/handlers.py

```diff
@@ -0,0 +1,179 @@
+import datetime
+import httpx
+import logging
+import stamina
+import app.actions.client as client
+
+from app.actions.configurations import AuthenticateConfig, FetchSamplesConfig, PullObservationsConfig
+from app.services.activity_logger import activity_logger
+from app.services.gundi import send_observations_to_gundi
+from app.services.state import IntegrationStateManager
+
+
+logger = logging.getLogger(__name__)
+
+
+state_manager = IntegrationStateManager()
+
+
+async def filter_and_transform(devices, integration_id, action_id):
+    def transform(device):
+        device_id = device.pop("Imeino")
+        device_name = device.pop("Vehicle_Name")
+
+        recorded_at = datetime.datetime.strptime(
+            device.pop("GPSActualTime"),
+            '%d-%m-%Y %H:%M:%S'
+        ).replace(tzinfo=datetime.timezone.utc)
+        lat = device.pop("Latitude")
+        lon = device.pop("Longitude")
+
+        additional = {k: v for k, v in device.items() if v and v != "--"}
```

Comment:
To avoid pop, what do you think about doing something like this:
```suggestion
        base_fields = ["Imeino", "Vehicle_Name", "GPSActualTime", "Latitude", "Longitude"]
        additional = {k: v for k, v in device.items() if k not in base_fields and v and v != "--"}
```